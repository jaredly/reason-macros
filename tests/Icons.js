// Generated by BUCKLESCRIPT VERSION 4.0.1100, PLEASE EDIT WITH CARE
'use strict';


var Link = /* module */[
  /* name */"IosLink",
  /* hi */"awesomehi"
];

var Camera = /* module */[
  /* name */"MdCamera",
  /* ho */"awesomeho"
];

exports.Link = Link;
exports.Camera = Camera;
/* No side effect */
