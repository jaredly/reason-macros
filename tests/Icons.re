///
///
// ok, yeah we're having to
// - somehow indicate that `Name` should be substituted with the eval'd `name`
//   - yeah let's do `Eval__${the name of the eval-time variable}`
// - the string literal in the external needs interpolation
//   - again, let's use the `eval` moniker, so `some$eval{name}`
//   - could also do `some$eval{uppercase(name)}` or something
// - for other identifiers, the `eval__` prefix will do the same thing.

let%macro.toplevel ionicon = (name: capIdent, iconName: capIdent) => {
  [%str module Eval__name = {
    [@bs.module] [@react.component]
    external make:
      (
        ~className: string=?,
        ~fontSize: string=?,
        ~color: string=?,
        ~onClick: 'event => unit=?
      ) =>
      React.element =
      "react-ionicons/lib/$eval{iconName}";
  }];
};

[%ionicon (Link, IosLink)];
[%ionicon (Camera, MdCamera)];
