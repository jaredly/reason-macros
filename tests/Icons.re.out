module Link = {
  let name = "IosLink";
  let hi = "awesomehi";
  [@bs.module] [@react.component]
  external make:
    (
      ~className: string=?,
      ~fontSize: string=?,
      ~color: string=?,
      ~onClick: 'event => unit=?
    ) =>
    string =
    "react-ionicons/lib/IosLink";
};
module Camera = {
  let name = "MdCamera";
  let ho = "awesomeho";
  [@bs.module] [@react.component]
  external make:
    (
      ~className: string=?,
      ~fontSize: string=?,
      ~color: string=?,
      ~onClick: 'event => unit=?
    ) =>
    string =
    "react-ionicons/lib/MdCamera";
};
